metrics/exporter.go

```diff
@@ -64,13 +64,54 @@ func newMetricsExporter(config *metricsConfig, logger *zap.SugaredLogger) error
 	return nil
 }
 
+func getKnativeRevisionMonitoredResource(gm *gcpMetadata) func(v *view.View, tags []tag.Tag) ([]tag.Tag, monitoredresource.Interface) {
+	return func(v *view.View, tags []tag.Tag) ([]tag.Tag, monitoredresource.Interface) {
+		kr := &KnativeRevision{
+			// The first three resource labels are from metadata.
+			Project:     gm.project,
+			Location:    gm.location,
+			ClusterName: gm.cluster,
+			// The rest resource labels are from metrics labels.
+			NamespaceName:     getResourceLabelValue(metricskey.LabelNamespaceName, tags),
+			ServiceName:       getResourceLabelValue(metricskey.LabelServiceName, tags),
+			ConfigurationName: getResourceLabelValue(metricskey.LabelConfigurationName, tags),
+			RevisionName:      getResourceLabelValue(metricskey.LabelRevisionName, tags),
+		}
+
+		var newTags []tag.Tag
+		for _, t := range tags {
+			// Keep the metrics labels that are not resource labels
+			fmt.Println(t.Key.Name() + ":" + t.Value)
+			if _, ok := metricskey.KnativeRevisionLabels[t.Key.Name()]; !ok {
+				newTags = append(newTags, t)
+			}
+		}
+
+		return newTags, kr
+	}
+}
+
+func getResourceLabelValue(key string, tags []tag.Tag) string {
```

Comment:
I wonder if it would be more efficient to get all these four values in one iteration. something like:
```
func updateResourceLabels(kr *KnativeRevision, tags []tag.Tag) {
	for _, t := range tags {
                switch t.Key,.Name() {
                case metricskey.LabelNamespaceName:
                    kr.NamespaceName = t.Value
                case metricskey.LabelRevisionName:
                    kr.RevisionName = t.Value
                case ...:
...
                }
	}
}
```

Reply:
Changed to use a map since we need to handle key missing case.